projects/wvr-elements/src/lib/shared/utility/bootstrap.utility.ts

```diff
@@ -85,7 +86,7 @@ const registerWeaverElements = (injector: Injector, wvrElements: Array<WvrElemen
             div.setAttribute('element', wvrElement.selector);
             const template = document.createElement('template');
             div.appendChild(template);
-            element.parentNode.replaceChild(div, element);
+            element.insertAdjacentElement('beforebegin', div);
```

Comment:
This seems to only insert element and not remove the one meant to replace.

Reply:
Correct, and by putting it on the DOM, we effectively cause the subsequence `appendChild()` that is on line 89/90 to move the element rather than do a remove and create.

See: https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild

```
...If the given child is a reference to an existing node in the document, appendChild() moves it from its current position to the new position (there is no requirement to remove the node from its parent node before appending it to some other node)...
...This means that a node can't be in two points of the document simultaneously. So if the node already has a parent, the node is first removed, then appended at the new position...
```

The wording on that is saying "removed and then appended", but it still exists on the DOM and so the destruction events never fire. This makes it effectively a "Move" operation.

Reply:
Oh thats cool.